xml/src/main/java/org/ehcache/xml/multi/XmlMultiConfiguration.java

```diff
@@ -0,0 +1,495 @@
+      ).collect(toMap(Map.Entry::getKey, Map.Entry::getValue));
+  }
+
+  /**
+   * Return this configuration as an XML {@link org.w3c.dom.Document}.
+   *
+   * @return configuration XML DOM.
+   */
+  public Document asDocument() {
+    return document;
+  }
+
+  /**
+   * Return this configuration as a rendered XML string.
+   *
+   * @return configuration XML string
+   */
+  public String asRenderedDocument() {
+    return renderedDocument;
+  }
+
+  @Override
+  public String toString() {
+    return asRenderedDocument();
+  }
+
+  private static Element unparseEhcacheConfiguration(Configuration config) {
+    if (config instanceof XmlConfiguration) {
+      return ((XmlConfiguration) config).asDocument().getDocumentElement();
+    } else {
+      return new XmlConfiguration(config).asDocument().getDocumentElement();
+    }
+  }
+
+  private interface Config {
+
+    Configuration configuration() throws IllegalStateException;
+
+    Configuration configuration(String variant);
+
+    Collection<Object> unparse(ObjectFactory factory);
+  }
+
+  private static class SingleConfig implements Config {
+
+    private final Configuration config;
+
+    private SingleConfig(Configuration config) {
+      this.config = config;
+    }
+
+    @Override
+    public Configuration configuration() {
+      return config;
+    }
+
+    @Override
+    public Configuration configuration(String variant) {
+      return configuration();
+    }
+
+    @Override
+    public Collection<Object> unparse(ObjectFactory factory) {
+      return singleton(unparseEhcacheConfiguration(config));
+    }
+  }
+
+  private static class VariantConfig implements Config {
+
+    private final Map<String, Configuration> configs;
+
+    private VariantConfig(Map<String, Configuration> configs) {
+      this.configs = configs;
+    }
+
+    @Override
+    public Configuration configuration() {
+      switch (configs.size()) {
+        case 0:
+          return null;
+        case 1:
+          return configs.values().iterator().next();
+        default:
+          throw new IllegalStateException("Please choose a variant: " + configs.keySet());
+      }
+    }
+
+    @Override
+    public Configuration configuration(String variant) {
+      Configuration configuration = configs.get(variant);
+      if (configuration == null) {
+        throw new IllegalArgumentException("Please choose a valid variant: " + configs.keySet());
+      } else {
+        return configuration;
+      }
+    }
+
+    @Override
+    public Collection<Object> unparse(ObjectFactory factory) {
+      return configs.entrySet().stream()
+        .map(v -> factory.createConfigurationsConfigurationVariant().withType(v.getKey())
+          .withConfig(unparseEhcacheConfiguration(v.getValue())))
+        .map(factory::createConfigurationsConfigurationVariant)
+        .collect(toList());
+    }
+  }
+
+  /**
+   * Create a builder seeded from an XML configuration.
+   * <p>
+   *
```

Comment:
Did you forget something in this "empty" paragraph?

Reply:
No... unless we decide to add something else while review the classloader documentation issue.